src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/HtmlSmartIndentFormatOnTypeProvider.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    internal class HtmlSmartIndentFormatOnTypeProvider : RazorFormatOnTypeProvider
+    {
+        public override string TriggerCharacter => "\n";
+
+        public override bool TryFormatOnType(Position position, FormattingContext context, out TextEdit[] edits)
+        {
+            if (position is null)
+            {
+                throw new ArgumentNullException(nameof(position));
+            }
+
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (!context.Options.TryGetValue(LanguageServerConstants.ExpectsCursorPlaceholderKey, out var value) && value.IsBool)
```

Comment:
This isn't complete. Also need to check the actual value of the property.

Reply:
Hmmm, this is how it's done in the other format providers 😆

Reply:
See [this](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CloseTextTagFormatOnTypeProvider.cs#L54) and [this](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CloseTextTagFormatOnTypeProvider.cs#L72)